tests/run_tests.sh

```diff
@@ -373,9 +377,10 @@ tree1_tests_rw () {
 check_tests () {
     tree="$1"
     shift
+    dtb=$BUILDDIR/$(basename "$tree").test.dtb
```

Comment:
I'm not quite understanding what's making `check_tests()` special here.  There are plenty of other places we build dtbs in the tests (every `run_dtc_test` for example).  In most cases, AFAICT, the approach taken is to retrieve the source from the base directory with an explicit `$SRCDIR` while using a relative path for the output, so expecting the script to be running in the build directory.

Why is that not working for these specific cases?  Or is it not working anywhere, in which case it needs to be fixed more widely as well.

Reply:
Ah I had not connected that run_tests.sh was running in the build directory. So the problem then is just that it's creating the dtb by taking "$SRCDIR/mydevices.dts" and appending ".test.dtb" to it, so the output file ends up in the $SRCDIR.

I don't need to add a $BUILDDIR variable then, rather I can do `dtb=$(basename "$tree" .dts).test.dtb` to have the same result as the other tests.

Reply:
Pushed updated version without $BUILDDIR

Reply:
Ah, much nicer, thanks!.